### PR TITLE
[Issue 165] Add init container docker image for HMS.

### PR DESCRIPTION
**Pull request by KenFigueiredo**
### Added
- [Issue-165](https://github.com/ExpediaGroup/apiary-data-lake/issues/165) Add init container dockerfile for supporting air-gapped environments.

Requirement for: https://github.com/ExpediaGroup/apiary-data-lake/issues/165

### CODE REVIEW

**Comment by massdosage** [comment]
> 
> 
> Looks good! Could you add an `initContainer/README.md` to explain a little about why this exists, how it is started (maybe link back to the terraform code where it is used), etc.?
> 
> @massdosage - what will it take to get a Jenkins job in EG setup for this (or changes to the current one) so that the init container is built alongside the HMS container?

Depends what you need to do ;) Happy to help.

**Comment by KenFigueiredo** [comment]
@barnharts4 Added Readme.

@massdosage - This would just need to be published & built alongside the Apiary docker image as this would become apart of the terraform deployable. Happy to help on the Jenkins setup for this as well!

File: initContainer/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Apiary Init Container
+
+This container was introduced as apart of resolving [[Issue 165](https://github.com/ExpediaGroup/apiary-data-lake/issues/165)].
```
Comment (barnharts4) [inline initContainer/README.md:3]:
```suggestion
This container was introduced as a part of resolving [[Issue 165](https://github.com/ExpediaGroup/apiary-data-lake/issues/165)]. 
```

File: initContainer/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+This solution relies on either [Kubernetes Init Containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) or [ECS Container dependencies](https://aws.amazon.com/about-aws/whats-new/2019/03/amazon-ecs-introduces-enhanced-container-dependency-management/), depending on your deployment target.
+
+
+The init docker container contains the same shell scripts which were previously ran as apart of the apiary terraform process.
+
+
+## Terraform Usages
+
+- Kubernetes
+  - [Read-Only](https://github.com/ExpediaGroup/apiary-data-lake/blob/9881ef5b6fdb37bd22ca132542104f7eb026ce0c/k8s-readonly.tf#L40)
```
Comment (barnharts4) [inline initContainer/README.md:17]:
```suggestion
  - [Read-Only](https://github.com/ExpediaGroup/apiary-data-lake/blob/master/k8s-readonly.tf#L40)
```
ditto for next 3 lines

File: initContainer/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+The general idea of why this container exists was to remove external dependencies and networking requirements between a deployment server and the deployment target when deploying Apiary. 
+
+Previously, Apiary had a hard requirement that the location of where the terraform apply job was running, had to have direct network access to the MySQL database in order to properly configure account access for the read-only and read-write roles. This presented some challenges when deploying into air-gapped environments as there would be no direct network access without punching a hole or proxying a connection between the deployer and the deployment environment.
+
+This solution relies on either [Kubernetes Init Containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) or [ECS Container dependencies](https://aws.amazon.com/about-aws/whats-new/2019/03/amazon-ecs-introduces-enhanced-container-dependency-management/), depending on your deployment target.
+
+
+The init docker container contains the same shell scripts which were previously ran as apart of the apiary terraform process.
```
Comment (barnharts4) [inline initContainer/README.md:11]:
```suggestion
The init docker container contains the same shell scripts which were previously run as a part of the apiary terraform process.
```

**Pull request merged**